trin-core/src/portalnet/overlay.rs

```diff
@@ -256,3 +261,61 @@ impl OverlayProtocol {
         }
     }
 }
+
+fn validate_find_nodes_distances(distances: &Vec<u16>) -> Result<(), OverlayRequestError> {
+    if distances.len() == 0 {
+        return Err(OverlayRequestError::InvalidRequest(
+            "Invalid distances: Empty list".to_string(),
+        ));
+    }
+    if distances.len() > 256 {
+        return Err(OverlayRequestError::InvalidRequest(
+            "Invalid distances: More than 256 elements".to_string(),
+        ));
+    }
+    let invalid_distances: Vec<&u16> = distances.iter().filter(|val| val > &&256u16).collect();
+    if invalid_distances.len() > 0 {
+        return Err(OverlayRequestError::InvalidRequest(format!(
+            "Invalid distances: Distances greater than 256 are not allowed. Found: {:?}",
+            invalid_distances
+        )));
+    }
+    let unique: HashSet<u16> = HashSet::from_iter(distances.iter().cloned());
+    if unique.len() != distances.len() {
+        return Err(OverlayRequestError::InvalidRequest(
+            "Invalid distances: Duplicate elements detected".to_string(),
+        ));
+    }
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use rstest::rstest;
+
+    #[rstest]
+    #[case(vec![0u16])]
+    #[case(vec![256u16])]
+    #[case((0u16..256u16).collect())]
+    fn test_nodes_validator_accepts_valid_input(#[case] input: Vec<u16>) {
+        let result = validate_find_nodes_distances(&input);
+        match result {
+            Ok(val) => assert_eq!(val, ()),
+            Err(_) => panic!("Valid test case fails"),
+        }
+    }
+
+    #[rstest]
+    #[case(vec![], "Empty list")]
+    #[case(vec![0u16; 257], "More than 256")]
```

Comment:
Technically this would also fail the duplicate test (as would anything array over length 256, I suppose). I'm fine leaving it as-is. Maybe a comment that it's reasonable to assume we test length first, since that's a faster check than duplicates.